Question2/UniqueCharacters.java

```diff
@@ -27,27 +26,28 @@ public int CountUniqueCharacters(String string) {
 		try {
 			if (string == null) {
 				throw new NullPointerException("String can't be null");
-			}
-			string = string.toLowerCase();
-			if (cache.containsKey(string)) {
-				return cache.get(string);
+			} else if (string == " ") {

```

Comment:
This is not properly checking for an empty string.Use !string.trim().isEmpty() method to check empty string.